src/c_api/c_predict_api.cc

```diff
@@ -243,6 +249,95 @@ int MXPredCreatePartialOut(const char* symbol_json_str,
   API_END_HANDLE_ERROR(delete ret);
 }
 
+int MXPredReshape(mx_uint num_input_nodes,
```

Comment:
when does this get called? during training or during setup?

Reply:
during testing, when you need to change the input shape of an already created predicator.